packages/hdwallet-walletconnect/src/ethereum.ts

```diff
@@ -0,0 +1,130 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import { ETHSignedMessage } from "@shapeshiftoss/hdwallet-core";
+
+interface JsonRpcResponse {
+  id: number
+  /**
+   * JSON-RPC Version
+   */
+  jsonrpc: "2.0"
+  /**
+   * Transaction Hash
+   */
+  result: any
+}
+
+export function describeETHPath(path: core.BIP32Path): core.PathDescription {
+  const pathStr = core.addressNListToBIP32(path);
+  const unknown: core.PathDescription = {
+    verbose: pathStr,
+    coin: "Ethereum",
+    isKnown: false,
+  };
+
+  if (path.length !== 5) return unknown;
+
+  if (path[0] !== 0x80000000 + 44) return unknown;
+
+  if (path[1] !== 0x80000000 + core.slip44ByCoin("Ethereum")) return unknown;
+
+  if ((path[2] & 0x80000000) >>> 0 !== 0x80000000) return unknown;
+
+  if (path[3] !== 0) return unknown;
+
+  if (path[4] !== 0) return unknown;
+
+  const index = path[2] & 0x7fffffff;
+  return {
+    verbose: `Ethereum Account #${index}`,
+    accountIdx: index,
+    wholeAccount: true,
+    coin: "Ethereum",
+    isKnown: true,
+  };
+}
+
+export async function ethVerifyMessage(_msg: core.ETHVerifyMessage, _ethereum: any): Promise<boolean | null> {
+  console.error("Method ethVerifyMessage unsupported for WalletConnect wallet!");
+  return null;
+}
+
+export function ethGetAccountPaths(msg: core.ETHGetAccountPath): Array<core.ETHAccountPath> {
+  const slip44 = core.slip44ByCoin(msg.coin);
+  if (slip44 === undefined) return [];
+  return [
+    {
+      addressNList: [0x80000000 + 44, 0x80000000 + slip44, 0x80000000 + msg.accountIdx, 0, 0],
+      hardenedPath: [0x80000000 + 44, 0x80000000 + slip44, 0x80000000 + msg.accountIdx],
+      relPath: [0, 0],
+      description: "WalletConnect",
+    },
+  ];
+}
+
+// TODO: implement sign tx
+export async function ethSignTx(msg: core.ETHSignTx, ethereum: any, from: string): Promise<core.ETHSignedTx | null> {
+  // const signedTx: JsonRpcResponse = 
+  return ethereum.request({
+    method: "eth_sendTransaction",
+    params: [msg],
+  });
+}
+
+export async function ethSendTx(msg: core.ETHSignTx, ethereum: any, from: string): Promise<core.ETHTxHash | null> {
+  try {
+    const utxBase = {
+      from: from,
+      to: msg.to,
+      value: msg.value,
+      data: msg.data,
+      chainId: msg.chainId,
+      nonce: msg.nonce,
+      // MetaMask, like other Web3 libraries, derives its transaction schema from Ethereum's official JSON-RPC API specification
```

Comment:
If this comment is relevant to WalletConnect as well, update the text to match:

```suggestion
      // WalletConnect, like other Web3 libraries, derives its transaction schema from Ethereum's official JSON-RPC API specification
```

Reply:
Pretty sure my entire `ethereum.ts` can be removed